src/collected_output.rs

```diff
@@ -54,15 +54,21 @@ impl Waiter {
         });
         let mut context_clone = context.clone();
         let capture_stderr = config.capture_stderr;
-        let stderr_join_handle = thread::spawn(move || -> io::Result<Vec<u8>> {
-            let mut collected_stderr = Vec::new();
+        let stderr_join_handle = thread::spawn(move || -> io::Result<Option<Vec<u8>>> {
```

Comment:
I wonder if this could be DRYed up. The code here seems nontrivial, so merging the stdout and stderr capture would be nice.

Reply:
What do you think of 4015e1c?

Reply:
Nice, looks good.